src/PrestaShopBundle/Routing/Linter/LegacyLinkLinter.php

```diff
@@ -0,0 +1,52 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShopBundle\Routing\Linter;
+
+use Symfony\Component\Routing\Route;
+
+/**
+ * Responsible for checking route _legacy_link configuration
+ */
+final class LegacyLinkLinter implements RouteLinterInterface
+{
+    /**
+     * Checks if _legacy_link is configured to route.
+     * Returns true if configured, false if not.
+     *
+     * @param Route $route
+     *
+     * @return bool
+     */
+    public function lint(Route $route)
+    {
+        if ($route->hasDefault('_legacy_link')) {
+            return true;
```

Comment:
@matks this is something I was thinking about. On error, should we return `false` or throw exception, wdyt? 

I see `lint()` as a command method thus it should follow _fire & forget_ approach (meaning no value should be returned) and throw exception only when error occurs (e.g. route is not configured with `_legacy_link`).

Reply:
I think `lint()` can rather be related to `isSomething()` functions.

`hasLangKey()` is expected to give us an information, rather true or false
`getLangKey()` is expected to retrieve an information for us, so it should either do its job and throw an exception if it is unable to do it

IMO the difference is that we *expect* some routes to be not valid, that's why we check them. Like validating a form versus validating a Command input in a handler. We expect user to sometimes input bad data, but we expect a Command to contain only valid data.

That's a very borderline issue 😅but this how I feel

Maybe `lint()` is not a good name, we should rather have `isLintValid()` and `performLint()` (this 2nd one would fix the code)

Reply:
Hmmm, ok `bool` would also work, but if we were to implement a chain of linters, then we wouldn't know which linter failed, because all linters would return just `false`, but by throwing exception, we would know exact problem. But again, this might be a borderline issue, so it's your call. :smile:

Reply:
You have a point. What about return `true` or an array of `$errors` ? This way we can stack errors and retrieve them all at once. Plus we might have several issues to return.

Reply:
Well, I don't like having different returns types (in this case `bool` and `array`). In addition, returing arrays doesn't make sense for single linter. :/

Reply:
Let's decide when we need to add a 2nd linter 😄 we will be able to modify this as this code is a debug/dev tool only, we dont need to follow SemVer here I think

Reply:
This is already the 3rd one. :smile: 

Second is here https://github.com/PrestaShop/PrestaShop/pull/13384 and the first one is already merged.